src/consts.ts

```diff
@@ -1,10 +1,12 @@
+export type NodeStatusValue = 'uninitialized' | 'loading' | 'done' | 'alert' | 'failed';
```

Comment:
1. Maybe `NodeStatus` will be sufficient and we don't need `Value` suffix?
2. I think you can use TS magic to avoid values duplication:
```
export type NodeStatus = typeof nodeStatuses[keyof typeof nodeStatuses];
```
When you add `as const` in L15, TS should provide the same type automatically for you without duplicating all the values 🙂 
Ref.: https://stackoverflow.com/questions/53662208/types-from-both-keys-and-values-of-object-in-typescript

Reply:
Ad 1. Yeah, you're right! I was fixed on the previous descriptions 😁 
Ad 2. I think that wouldn't be possible as in that case the `nodeStatuses` would be using `Record<string, NodeStatus>` type, which would reference to itself (by `NodeStatus`)
I've just tested it and it seems that it doesn't allow to do such a recurring magic 😁

Reply:
(I've added changes regarding the name of the `NodeStatusValue` type)

Reply:
Hmm... Can you check my patch?
<details>
<summary>Click to expand</summary>
Index: src/consts.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/consts.ts b/src/consts.ts
--- a/src/consts.ts	(revision ea433b7701efcf7e08779e19b21226121b82fcfb)
+++ b/src/consts.ts	(date 1644580967320)
@@ -1,17 +1,16 @@
-export type NodeStatus = 'uninitialized' | 'loading' | 'done' | 'alert' | 'failed';
-type NodeStatuses = Record<string, NodeStatus>;
+export type NodeStatus = typeof nodeStatuses[keyof typeof nodeStatuses];
 
 /**
  * Node statuses constants.
  *
  * @enum {string}
  */
-export const nodeStatuses: NodeStatuses = {
+export const nodeStatuses = {
     UNINITIALIZED: 'uninitialized',
     LOADING: 'loading',
     DONE: 'done',
     ALERT: 'alert',
     FAILED: 'failed'
-};
+} as const;
 
 export default { nodeStatuses };
Index: src/icons.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/icons.ts b/src/icons.ts
--- a/src/icons.ts	(revision ea433b7701efcf7e08779e19b21226121b82fcfb)
+++ b/src/icons.ts	(date 1644581682168)
@@ -49,16 +49,17 @@
  *
  * @param {NodeStatus} nodeStatus - node status value.
  *
- * @returns {string} character from cloudify font or empty string for not {@link nodeStatuses} value
+ * @returns {string} character from cloudify font for {@link NodeStatus} values (expect {@link nodeStatuses.UNINITIALIZED}) or empty string for other values
  */
 export function getNodeStatusIcon(nodeStatus: NodeStatus): string {
+    const defaultChar = '';
     const nodeStatusToChar = {
         [nodeStatuses.ALERT]: '\ue629',
         [nodeStatuses.DONE]: '\ue62a',
         [nodeStatuses.FAILED]: '\ue62b',
-        [nodeStatuses.LOADING]: '\ue630'
+        [nodeStatuses.LOADING]: '\ue630',
+        [nodeStatuses.UNINITIALIZED]: defaultChar
     };
-    const defaultChar = '';
     const knownType = !!nodeStatusToChar[nodeStatus];
 
     return knownType ? nodeStatusToChar[nodeStatus] : defaultChar;

</details>

Reply:
I've double checked that and it seems to be working flawlessly

I must've been in hurry, before going on the vacations 😁 
Thanks for providing the patch, now the code is as automatic as it may be 😎